pallets/oracle-data-collection/src/lib.rs

```diff
@@ -460,17 +541,58 @@ pub mod types {
 		}
 	}
 
+	/// Information of a collection
+	#[derive(
+		Encode, Decode, PartialEq, Eq, Clone, TypeInfo, RuntimeDebugNoBound, MaxEncodedLen,
+	)]
+	#[scale_info(skip_type_params(T))]
+	pub struct CollectionInfo<T: Config> {
+		/// Maximum duration to consider an oracle value non-outdated.
+		/// An oracle value is consider updated if its timestamp is higher
+		/// than `now() - value_lifetime`
+		pub value_lifetime: Option<T::Timestamp>,
+
+		/// Minimun number of feeders to succesfully aggregate a value.
+		pub min_feeders: u32,
+	}
+
+	impl<T: Config> Default for CollectionInfo<T> {
+		fn default() -> Self {
+			Self {
+				value_lifetime: None,
+				min_feeders: 0,
+			}
+		}
+	}
+
 	/// A collection cached in memory
-	#[derive(Clone)]
-	pub struct CachedCollection<T: Config>(
-		pub BoundedBTreeMap<T::OracleKey, OracleValuePair<T>, T::MaxCollectionSize>,
-	);
+	#[derive(Encode, Decode, Clone, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+	#[scale_info(skip_type_params(T))]
+	pub struct CachedCollection<T: Config> {
+		pub content: BoundedBTreeMap<T::OracleKey, OracleValuePair<T>, T::MaxCollectionSize>,
+		pub older_timestamp: T::Timestamp,
```

Comment:
NIT: Naming `older_timestamp` confuses me.

Reply:
Any suggestion to indicate the oldest timestamp in the collection?

Reply:
Last_updated maybe?

Reply:
I understand `last_updated` as the timestamp of the last time the collection was updated. But we want to express that this indicates the timestamp of the older oracle value in the collection. It's difficult to find an appropriate name for it 😆

Reply:
Maybe `older_value_timestamp` would better express that it belongs to a value of the collection instead of the whole collection?

Reply:
Fine for me